ui/components/DataTable.tsx

```diff
@@ -165,8 +168,14 @@ function UnstyledDataTable({
           <TableHead>
             <TableRow>
               {_.map(fields, (f) => (
-                <TableCell key={f.label}>
-                  <SortableLabel field={f} />
+                <TableCell
+                  key={typeof f.label === "function" ? f.label(r) : r[f.label]}
```

Comment:
I'm not sure how `key` responds to receiving a React component? 🤔 ..

```ts
    type Key = string | number;
```

Maybe we still require a `label: string` that could be `"checkbox"` for example, but also provide a `labelRenderer` or something too that can return a react component like `labelRenderer ? labelRenderer(r) : label` wdyt?

Reply:
Sounds v good!